src/misc/why_kotlin/why_kotlin.md

```diff
@@ -127,7 +128,29 @@ fun nullSafeMethodCall(a: Int?): Double {
 ```
 
 We'll combine the concept above with the `?.` operator, which performs a null safe method call. 
-If a is null, Kotlin won't try to call `.toDouble()` on it and will just return null, which will then be caught by the Elvis operator!
+If `a` is null, Kotlin won't try to call `.toDouble()` on it and will just return null,
+which will then be caught by the Elvis operator!
+nsure that you access your arm variable in init in some way to trigger the by lazy function.
+### Accessing Hardware
+Because of Kotlin's null safety system, accessing hardware must be done differently.
+There are a few ways to do this, but the one we'll show here is to use `by lazy`:
+
+```kt
+    // by lazy will only initialize this variable the first time it is used.
+    // This prevents it from ever being null, but also allows you to initialize it only after your opmode begins.
+    val arm by lazy { hardwareMap["arm"] as DcMotorEx } // Alternately hardwareMap.get(DcMotorEx::class.java, "arm") also works here
+
+    override fun init() { // or runOpMode() for LinearOpModes
+        // To ensure that your hardware is initialized in the init stage, access it in some way to trigger the by lazy function
+        arm
```

Comment:
note that in this example you already initialize it when you call `arm.power`
maybe make this more clear, that's about it

Reply:
I just decided to remove the whole initialization example. This would slow down the first loop in some scenarios, but it would also ensure G405 compliance if the user isn't accessing in init. Kind of uncertain about this change though.